src/apps/the-exchange/components/EnterAmount/EnterAmount.tsx

```diff
@@ -29,72 +62,155 @@ const EnterAmount = ({ type, tokenSymbol }: EnterAmountProps) => {
             toTokenAddress: receiveToken?.address || '',
             toChainId: receiveToken?.chainId || 0,
             toTokenDecimals: receiveToken?.decimals || 0,
+            slippage: 0.05,
+        };
+
+        try {
+            const offer = await getBestOffer(params);
+
+            if (offer && receiveToken) {
+                const usdPrice = await getPrice(receiveToken.address, receiveToken.chainId);
+                if (usdPrice) {
+                    setAmountReceive({
+                        tokenAmount: offer.tokenAmountToReceive,
+                        usdAmount: usdPrice.usd * offer.tokenAmountToReceive,
+                    });
+                } else {
+                    setAmountReceive({ tokenAmount: offer.tokenAmountToReceive, usdAmount: 0 });
+                }
+            } else {
+                setIsNoOffer(true);
+            }
+
+            return offer;
+        } catch (error) {
+            console.error('Failed to fetch the best offer');
+            return null;
+        } finally {
+            setIsOfferLoading(false);
         }
```

Comment:
So you have quite a large try catch here - whilst this is not wrong, personally i try to use try / catch as a last resort when it comes to error control. This is because promises have a built-in `.catch` method and then also because nothing "escapes" the try / catch which is why i use it as a last resort, and a `.catch` first - this gives you an extra level of error control.